lib/spack/spack/concretize.py

```diff
@@ -190,10 +190,49 @@ def _concretize_task(packed_arguments: Tuple[int, str, TestsType]) -> Tuple[int,
     index, spec_str, tests = packed_arguments
     with tty.SuppressOutput(msg_enabled=False):
         start = time.time()
-        spec = Spec(spec_str).concretized(tests=tests)
+        spec = concretized(Spec(spec_str), tests=tests)
         return index, spec, time.time() - start
 
 
+def concretized(spec: Spec, tests: Union[bool, Iterable[str]] = False) -> Spec:
+    """Return a concretized copy of the given spec.
+
+    Args:
+        tests: if False disregard 'test' dependencies, if a list of names activate them for
+            the packages in the list, if True activate 'test' dependencies for all packages.
+    """
+    spec.replace_hash()
+
+    for node in spec.traverse():
+        if not node.name:
+            raise spack.error.SpecError(
+                f"Spec {node} has no name; cannot concretize an anonymous spec"
+            )
+
+    if spec._concrete:
+        return spec.copy()
+
+    allow_deprecated = spack.config.get("config:deprecated", False)
+    solver = spack.solver.asp.Solver()
```

Comment:
Missing `import spack.solver.asp`.

But can be made a top-level import with my other suggestion.